tests/main/snapd-apparmor/task.yaml

```
@@ -0,0 +1,73 @@
+summary: Ensure snapd-apparmor works as expected
+
+environment:
+    CONSUMER_SNAP: test-snapd-policy-app-consumer
+
+debug: |
+    journalctl -u snap.apparmor.service
+
+execute: |
+    if ! systemctl is-active snapd.apparmor.service; then
+      echo "Skipping test since snapd.apparmor.service is not active"
+      exit 0
+    fi
+
+    echo "Ensure snapd.apparmor is enabled"
+    systemctl is-enabled snapd.apparmor.service
+
+    # install a test snap which generates a lot of apparmor policies
+    echo "Given a test snap is installed"
+    "$TESTSTOOLS"/snaps-state install-local "$CONSUMER_SNAP"
+    tests.cleanup defer snap remove --purge "$CONSUMER_SNAP"
+
+    # stop snapd so it does not try and load apparmor policies directly
```

snapd will already have loaded policies as part of installing the snap, maybe

```suggestion
    # stop snapd so we can unload the apparmor policies manually
```